### ISSUE TITLE
IOS: RadListView srollToIndex error

### ISSUE THREAD

**Issue by Cheas** [NativeScript/nativescript-ui-feedback#1215]
> Please take a minute to read our [NativeScript Code of Conduct](https://github.com/NativeScript/codeofconduct) before proceeding with posting issues or discussing. The purpose of this guide is to make communication and cooperation within our forums a pleasure for you and the other members.

### Tell us about the problem
I use in the project component RadlistView and function scrollToIndex (6 different modules).
Everywhere an error occurs - 

this.items.getItem is not a function. (In 'this.items.getItem(index)', 'this.items.getItem' is undefined) 

### Which platform(s) does your issue occur on?
Android - ok, IOS - errors

### Please provide the following version numbers that your issue occurs with:
OK
----------------------------------
CLI: 4.2 Angular  
"nativescript-ui-listview": "^3.8.0",
"tns-core-modules": "^4.2.1",
"tns-ios":  "4.2.0"

Errors
----------------------------------
 CLI: 5.4 Angular  
"nativescript-ui-listview": "^6.2.0",
"tns-core-modules": "^5.4.3",
"tns-ios":  "4.2.0"   

 CLI: 6.0 Angular 
"nativescript-ui-listview": "^7.0.4",
"tns-core-modules": "^6.0.6",
"tns-ios":  "6.0.2"  

### Please tell us how to recreate the issue in as much detail as possible.

This situation occurs for any index values (real or not). The timeout does not help, setting the height of the elements too.

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
HTML
```
<RadListView #radListView row="10" [items]="listItems" [itemTemplateSelector]="templateSelector" (itemLoading)="onItemLoading($event)" (scrolled)="onScrolled($event)" [visibility]="lb && status == 3? 'visible' : 'collapsed'">
    <ng-template tkTemplateKey="row" let-player="item">
        <GridLayout class="player-item" [class.is-current]="player && player.isCurrentUser" columns="10,48,5,auto,10,*,5,auto,10" rows="*" verticalAlignment="center">
            <Label row="0" col="1" [text]="player ? player.detailRank : ''" class="text-center" verticalAlignment="center"></Label>
            <StackLayout row="0" col="3" verticalAlignment="center" class="player-avatar" (tap)="onTapPlayerInfo($event,player)">
                <avatar-image [firstName]="player ? player.firstName : ''" [lastName]="player ? player.lastName : ''" [avatarPath]="player ? player.avatarPath : ''">
                </avatar-image>
            </StackLayout>
            <Label row="0" col="5" class="player-name" [text]="player ? player.firstName+' '+player.lastName : ''" verticalAlignment="center"></Label>
            <Label row="0" col="7" [text]="player ? player.detailStr : ''" verticalAlignment="center"></Label>
        </GridLayout>
    </ng-template>
</RadListView>
```
TS
```
if ((index > 0) && (index < this.listItems.length - 1)) {
    this.radListViewComponent.listView.scrollToIndex(index, false, ListViewItemSnapMode.Start)
}
```

**Comment by tsonevn** [comment]
HI @Cheas, 
I tested the described case in a sample project while using latest {N} `6.0` and `nativescript-ui-listview` `7.0.4`, however, was unable to recreate the error. For your convenience, I am attaching a sample project, please review it and make the needed changes, that will allow us to recreate the problem.
[Archive.zip](https://github.com/NativeScript/nativescript-ui-feedback/files/3539722/Archive.zip)

**Comment by Cheas** [comment]
Hi. Nikolay. Thank you for your attention to my problems.

1. I checked your project, everything works.
2. Integrated your project into my project. Your part is working, mine is
not working.
3. Compared the conditions for the component RadListView, filtering and
sorting is used for mine. If I remove the sort sorting and filtering, then
everything works in my project too.
It all worked with nativescript 4.2.0 &  nativescript-ui-listview 3.8.0.


I made changes to your project to illustrate the problem (link):
a. launch the application
b. tap 'Scrolling to 4' - ok
c. tap 'Filtering' or ' 'Sorting'
d. tap 'Scrolling to 4' - error

link -
https://drive.google.com/file/d/1a0XyPbP1CyEReuLUT2k2N7tiLOqTlTKs/view

Alexandr.

пн, 26 авг. 2019 г. в 09:38, Nikolay Tsonev <notifications@github.com>:

**Comment by tsonevn** [comment]
HI @Cheas,
I've checked the provided project and was able to recreate the problem. I will mark it as a bug and we will investigate further the case.

**Comment by tsonevn** [comment]
Hi @Cheas,
We made some further investigation on your case and found out that this issue happens when a pure-JavaScript array is used for the RadListView's items source. On that matter as a workaround, you can replace the **Array** with **ObservableArray**, provided by `tns-core-modules`. For example:
```
import {ObservableArray} from "tns-core-modules/data/observable-array";
....
private items = new ObservableArray<Item>(
        { id: 1, name: "Ter Stegen", role: "Goalkeeper" },
        { id: 3, name: "Piqué", role: "Defender" },
        { id: 4, name: "I. Rakitic", role: "Midfielder" },
        { id: 5, name: "Sergio", role: "Midfielder" },
        { id: 6, name: "Denis Suárez", role: "Midfielder" },
        { id: 7, name: "Arda", role: "Midfielder" },
        { id: 8, name: "A. Iniesta", role: "Midfielder" },
        { id: 9, name: "Suárez", role: "Forward" },
        { id: 10, name: "Messi", role: "Forward" },
        { id: 11, name: "Neymar", role: "Forward" },
        { id: 12, name: "Rafinha", role: "Midfielder" },
        { id: 13, name: "Cillessen", role: "Goalkeeper" },
        { id: 14, name: "Mascherano", role: "Defender" },
        { id: 17, name: "Paco Alcácer", role: "Forward" },
        { id: 18, name: "Jordi Alba", role: "Defender" },
        { id: 19, name: "Digne", role: "Defender" },
        { id: 20, name: "Sergi Roberto", role: "Midfielder" },
        { id: 21, name: "André Gomes", role: "Midfielder" },
        { id: 22, name: "Aleix Vidal", role: "Midfielder" },
        { id: 23, name: "Umtiti", role: "Defender" },
        { id: 24, name: "Mathieu", role: "Defender" },
        { id: 25, name: "Masip", role: "Goalkeeper" }
    );

    getItems(): ObservableArray<Item> {
        return this.items;
    }
....
```

**Comment by Cheas** [comment]
Hi, NikolaY. I checked this option with your simple example - everything
really works. It takes more time to apply this to my project, but I hope
that everything will be fine. Thank.

вт, 3 сент. 2019 г. в 11:51, Nikolay Tsonev <notifications@github.com>:

**Comment by zbranzov** [comment]
A fix is released in nativescript-ui-listview@7.0.5
